### ISSUE TITLE
Format percent gives FunctionClauseError

### ISSUE THREAD

**Issue by maennchen** [elixir-cldr/cldr_numbers#13]
## Code

```elixir
@client.risk_profile.annual_return_expectation
|> Acme.Cldr.Number.to_string!(format: :percent, fractional_digits: 1)
```

## Error

```
** (FunctionClauseError) no function clause matching in Cldr.Number.to_string/4

The following arguments were given to Cldr.Number.to_string/4:

    # 1
    0.05

    # 2
    :percent

    # 3
    Acme.Cldr

    # 4
    %Cldr.Number.Format.Options{currency: nil, currency_digits: :accounting, currency_spacing: nil, currency_symbol: nil, format: :percent, fractional_digits: 1, locale: %Cldr.LanguageTag{backend: Ac.Cldr, canonical_locale_name: "de-Latn-CH", cldr_locale_name: "de-CH", extensions: %{}, gettext_locale_name: nil, language: "de", language_subtags: [], language_variant: nil, locale: %{}, private_use: [], rbnf_locale_name: "de-CH", requested_locale_name: "de-CH", script: "Latn", territory: :CH, transform: %{}}, minimum_grouping_digits: 0, number_system: :latn, pattern: :positive, round_nearest: nil, rounding_mode: :half_even, symbols: %Cldr.Number.Symbol{decimal: ".", exponential: "E", group: "’", infinity: "∞", list: ";", minus_sign: "-", nan: "NaN", per_mille: "‰", percent_sign: "%", plus_sign: "+", superscripting_exponent: "·", time_separator: ":"}}

Attempted function clauses (showing 10 out of 13):

    defp to_string(number, :ordinal, backend, %{locale: locale})
    defp to_string(number, :spellout, backend, %{locale: locale})
    defp to_string(number, :spellout_verbose, backend, options)
    defp to_string(number, :spellout_year, backend, options)
    defp to_string(number, :spellout_cardinal = format, backend, %{locale: locale})
    defp to_string(number, :spellout_ordinal = format, backend, options)
    defp to_string(number, :spellout_ordinal_verbose = format, backend, options)
    defp to_string(number, :roman, backend, _options)
    defp to_string(number, :roman_lower, backend, _options)
    defp to_string(number, :currency_long = format, backend, options)
    ...
    (3 clauses not shown)

code: case RiskProfileOnlyProfile.compile(
stacktrace:
  (ex_cldr_numbers 2.13.1) lib/cldr/number.ex:432: Cldr.Number.to_string/4
  (ex_cldr_numbers 2.13.1) lib/cldr/number.ex:373: Cldr.Number.to_string/3
  (ex_cldr_numbers 2.13.1) lib/cldr/number.ex:421: Cldr.Number.to_string!/3
  (acme 3.1.0) [PATH]/_build/test/lib/acme/priv/document/shared/risk_analysis.tex.eex:47: Acme.Document.Shared."_render_to_tex_risk_analysis.tex"/1
  (acme 3.1.0) [PATH]/_build/test/lib/acme/priv/document/risk_profile_only_profile/main.tex.eex:82: anonymous fn/4 in Acme.Document.RiskProfileOnlyProfile."_render_to_tex_main.tex"/1
  (elixir 1.10.1) lib/enum.ex:2111: Enum."-reduce/3-lists^foldl/2-0-"/3
  (acme 3.1.0) [PATH]/_build/test/lib/acme/priv/document/risk_profile_only_profile/main.tex.eex:81: Acme.Document.RiskProfileOnlyProfile."_render_to_tex_main.tex"/1
  (acme 3.1.0) lib/acme/document_generator.ex:167: Acme.Document.RiskProfileOnlyProfile.compile/1
  test/acme/document/risk_profile_only_profile_test.exs:58: (test)
```

## Versions

* `ex_cldr` - `2.14.1`
* `ex_cldr_numbers` - `2.13.1`

Back on `ex_cldr` `2.13.0` and `ex_cldr_numbers` `2.13.1 the error does not happen.

**Comment by kipcole9** [comment]
Sorry about this poor regression which is now fixed in [ex_cldr_numbers version 2.13.2](https://hex.pm/packages/ex_cldr_numbers/2.13.2).

### Changelog entry

* Fix regression and allow `:percent` formats in `Cldr.Number.to_string/3`. Thanks to @maennchen. Fixes #13.

### Cause of regression

I completely rewrote options handling for `Cldr.Number.to_string/3` and omitted `:percent` as a valid standard format (standard formats are resolved during options processing into the underlying format string).  And while there are a lot of tests for format strings with `%` tokens, there were none using the `:percent` format.  Test is now added.

**Comment by maennchen** [comment]
@kipcole9 As always: Thanks you very much, this is a great library 😊
